contracts/PoaOperator.sol

```diff
@@ -99,6 +99,17 @@ contract PoaOperator is Adminable {
     _setSlot(_slotId, _signerAddr, _tenderAddr);
   }
 
+  function setTakenSlotBitmap(uint256 _before, uint256 _pos, bool isActive) internal pure returns (uint256) {
+    if (isActive) {
+      return _before | (0x01 << (255 - _pos));
+    } else {
+      // if the bit is on, turn it off through XOR
+      if (_before | (0x01 << (255 - _pos)) > 0) {
```

Comment:
can be replaced with

```
return _before & ~(0x01 << (255 - _pos));
```

?

Reply:
1. (0x01 << (255 - _pos) — sets the bit e.g. (0x000100)
2. ~... — inverts the number so the bit is off (e.g. 0x111011)
3. .. &  .. — clears the bit in `_before`